cmd/registry-replacer/main.go

```diff
@@ -351,3 +358,20 @@ type censor struct {
 func (c *censor) Censor(data []byte) []byte {
 	return bytes.ReplaceAll(data, c.secret, []byte("<< REDACTED >>"))
 }
+
+// applyReplacementsToDockerfile duplicates what the build tools would do
+func applyReplacementsToDockerfile(in []byte, image *api.ProjectDirectoryImageBuildStepConfiguration) ([]byte, error) {
```

Comment:
`image` can be `nil`

Reply:
If that is the case we would never get to this code, we have a check in the very beginning:
```
if len(config.Images) == 0 {
    return nil
}
```